src/business-logic/services/test/device-service.spec.ts

```diff
@@ -0,0 +1,41 @@
+import {DeviceServiceImplementation} from '../device-service-implementation'
+import {Device, TelemetricsDevice} from '../../../model/entities/device'
+import {instance, mock, verify, when} from '@typestrong/ts-mockito'
+import {MongoDeviceRepository} from '../../../data-access/repositories/mongo/mongo-device-repository'
+import {DeviceService} from '../interfaces/device-service'
+import {DeviceEventEmitter} from '../interfaces/device-event-emitter'
+
+describe('DeviceService', () => {
+  let testDevice: TelemetricsDevice
+  let deviceRepository: MongoDeviceRepository
+  let deviceEventEmitter: DeviceEventEmitter
+  const mockedValue: MongoDeviceRepository = mock<MongoDeviceRepository>()
+
+  beforeEach(() => {
+    testDevice = mock<TelemetricsDevice>()
+    deviceRepository = instance(mockedValue)
+    deviceEventEmitter = mock<DeviceEventEmitter>()
+  })
+
+
+  it('should read all configurations', async () => {
+    const devices: Device[] = [testDevice]
+    const mockedService: DeviceService = mock<DeviceService>()
+    const testee: DeviceServiceImplementation = new DeviceServiceImplementation(deviceRepository, deviceEventEmitter)
+    when(mockedService.getDevices()).thenReturn(Promise.resolve(devices))
+
+    await testee.getDevices()
+
+    verify(mockedValue.getDevices()).called()
+  })
+
+  it('should create a new configuration', async () => {
+    const mockedService: DeviceService = mock<DeviceService>()
+    const testee: DeviceServiceImplementation = new DeviceServiceImplementation(deviceRepository, deviceEventEmitter)
+    when(mockedService.create(testDevice)).thenReturn(Promise.resolve())
```

Comment:
Line 33 and 35 has no effect on the test and can be removed.

A good rule of thumb: If you are mocking the interface of the class you are testing, then there is a good chance that you are mocking the wrong thing :)